src/views/users/use-add-groups.tsx

```diff
@@ -0,0 +1,30 @@
+import { useAuthAxios } from '../../hooks'
+import { useAlert } from '@dhis2/app-service-alerts'
+
+export const useAddGroups = () => {
+    const [{ loading: addGroupLoading }, addGroup] = useAuthAxios({ method: 'POST' }, { manual: true })
+
+    const { show: showAlert } = useAlert(
+        ({ message }) => message,
+        ({ isCritical }) => (isCritical ? { critical: true } : { success: true })
+    )
+
+    const addGroups = async (groups: string[], userId: number) => {
```

Comment:
This should be wrapped in a `useCallback`.

In this case it's not an issue because no other hook has a dependency on this function, which would require the function's reference to remain the same. But if that ever changes, it's a pain to debug.

I'd say it's good practice to always use `useCallback` for functions returned by hooks when meant for re-use (contrary to functions created within a component, then it depends on the usage and the code is not meant for re-use). This way the function can be used in any context